### PR TITLE
[Fix](Outfile) Support export nested complex type data to orc file format

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: regression-test/suites/export_p0/outfile/orc/test_outfile_orc_one_nested_type.groovy [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+    }
+
+    // 2. test NOT NULL STRUCT_STRUCT
+    try {
+        def struct_field_define = "`ss_info` STRUCT<s_info:STRUCT<s_id:int(11), s_name:string, s_address:string>> NOT NULL"
+        // create table to export data
+        create_table(export_table_name, struct_field_define)
+
+        // insert data
+        sql """ insert into ${export_table_name} values (1, 'doris1', {{1, 'sn1', 'sa1'}}); """
```
Comment (amorynan) [inline regression-test/suites/export_p0/outfile/orc/test_outfile_orc_one_nested_type.groovy:108]: we should avoid use struct literal with '{}' which is confused with map literal
And struct we only support with function creator, eg. struct(), name_struct() , old struct literal just keep  for old client, but we not suggest to use it.

File: be/src/vec/data_types/serde/data_type_array_serde.cpp [hunk 2]
```diff
@@ -361,24 +361,18 @@ Status DataTypeArraySerDe::write_column_to_orc(const std::string& timezone, cons
...
         size_t offset = offsets[row_id - 1];
         size_t next_offset = offsets[row_id];
-
-        if (cur_batch->notNull[row_id] == 1) {
-            static_cast<void>(nested_serde->write_column_to_orc(timezone, nested_column, nullptr,
-                                                                cur_batch->elements.get(), offset,
-                                                                next_offset, buffer_list));
-        }
-
+        static_cast<void>(nested_serde->write_column_to_orc(timezone, nested_column, nullptr,
```
Comment (amorynan) [inline be/src/vec/data_types/serde/data_type_array_serde.cpp:373]: why here not need to consider error for nested ?  if error happen , will not make orc content mistake?

File: be/src/vec/data_types/serde/data_type_map_serde.cpp [hunk 3]
```diff
@@ -483,29 +483,23 @@ Status DataTypeMapSerDe::write_column_to_orc(const std::string& timezone, const
...
-                                                             cur_batch->keys.get(), offset,
-                                                             next_offset, buffer_list));
-            static_cast<void>(value_serde->write_column_to_orc(timezone, nested_values_column,
-                                                               nullptr, cur_batch->elements.get(),
-                                                               offset, next_offset, buffer_list));
-        }
+        static_cast<void>(key_serde->write_column_to_orc(timezone, nested_keys_column, nullptr,
+                                                         cur_batch->keys.get(), offset, next_offset,
+                                                         buffer_list));
+        static_cast<void>(value_serde->write_column_to_orc(timezone, nested_values_column, nullptr,
```
Comment (amorynan) [inline be/src/vec/data_types/serde/data_type_map_serde.cpp:500]: same with array-serde , do not need consider error status?

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**